packages/react/src/utils/useFocusableWhenDisabled.ts

```diff
@@ -40,7 +40,11 @@ export function useFocusableWhenDisabled(
       additionalProps['aria-disabled'] = disabled;
     }
 
-    if (isNativeButton && !composite && !focusableWhenDisabled) {
+    if (
+      isNativeButton &&
+      ((!composite && focusableWhenDisabled !== true) ||
+        (composite && focusableWhenDisabled === false))
+    ) {
```

Comment:
I think it's worth extracting these to variables - the `boolean | undefined` value can be a bit confusing

> const focusableWhenDisabled: boolean | undefined
Whether the component should be focusable when disabled. When undefined, composite items are focusable when disabled by default.